### PR TITLE
Part 6: Last of azure-common-* checkstyle fixes

### PR DESCRIPTION
**Pull request by conniey**
Finishing last of test checkstyle fixes:
* azure-common-auth and azure-common-mgmt
  * Remove MockHttpResponse and use one from azure-common-test
* Adds a bunch of javadocs to test classes that don't fit the Mock* or *Test class. (I didn't want to add one off classes to checkstyle exclusions)
* Using getter/setters for test classes
* Fixing indentation
* Fixing name of static final variables

### CODE REVIEW

File: common/azure-common-auth/src/test/java/com/azure/common/auth/credentials/http/MockHttpClient.java [hunk 1]
```diff
@@ -17,13 +18,20 @@
...
+     */
     public MockHttpClient() {
         requests = new ArrayList<>();
     }
 
+    /**
+     * Gets the requests sent by MockHttpClient.
+     *
+     * @return The requests sent by this HTTP client.
+     */
```
Comment (JonathanGiles) [inline common/azure-common-auth/src/test/java/com/azure/common/auth/credentials/http/MockHttpClient.java]: We should remove the requirement that public unit test methods require JavaDoc.

Comment (conniey) [reply common/azure-common-auth/src/test/java/com/azure/common/auth/credentials/http/MockHttpClient.java]: I added a rule here that ignores Javadoc for classes that start with `Mock*`. Is that sufficient? There are existing rules that exclude `*Test`.
https://github.com/Azure/azure-sdk-for-java/pull/3412/files#diff-8ab8d3eeaecf610d7802a6417b5bcdf5R16

Comment (JonathanGiles) [reply common/azure-common-auth/src/test/java/com/azure/common/auth/credentials/http/MockHttpClient.java]: I would just apply it to anything under `/src/test/java`

Comment (conniey) [reply common/azure-common-auth/src/test/java/com/azure/common/auth/credentials/http/MockHttpClient.java]: Does this work? https://github.com/Azure/azure-sdk-for-java/pull/3412/files?file-filters%5B%5D=.xml#diff-8ab8d3eeaecf610d7802a6417b5bcdf5R17

I removed the ones that had overlapping logic.

**Pull request merged**